sugondat/shim/src/sugondat_rpc/conn.rs

```diff
@@ -9,6 +9,68 @@ use subxt::backend::rpc::RpcClient;
 use sugondat_subxt::sugondat::is_codegen_valid_for;
 use tokio::sync::{oneshot, Mutex};
 
+// Keeping the connection and connector inside an Arc is useful because
+// the same connection will be used for all the docks, and thus
+// it's shared among all of them
+#[derive(Clone)]
+pub enum ConnectionType {
```

Comment:
style: a "type" would likely be a non-data-carrying union

this is just a `Connection` (which has two different variants)